engine/consensus/sealing/engine.go

```diff
@@ -84,20 +83,24 @@ func NewEngine(log zerolog.Logger,
 	}
 
 	e := &Engine{
-		unit:                                 engine.NewUnit(),
-		log:                                  log.With().Str("engine", "sealing.Engine").Logger(),
-		me:                                   me,
-		engineMetrics:                        engineMetrics,
-		cacheMetrics:                         mempool,
-		headers:                              headers,
-		payloads:                             payloads,
-		requiredApprovalsForSealConstruction: options.RequiredApprovalsForSealConstruction,
-		rootHeader:                           rootHeader,
+		unit:          engine.NewUnit(),
+		log:           log.With().Str("engine", "sealing.Engine").Logger(),
+		me:            me,
+		engineMetrics: engineMetrics,
+		cacheMetrics:  mempool,
+		headers:       headers,
+		payloads:      payloads,
+		rootHeader:    rootHeader,
 	}
 
-	err = e.setupMessageHandler()
+	err = e.setupTrustedInboundQueues()
 	if err != nil {
-		return nil, fmt.Errorf("could not initialize message handler: %w", err)
+		return nil, fmt.Errorf("initialization of inbound queues for trusted inputs failed: %w", err)
+	}
+
+	err = e.setupMessageHandler(options.RequiredApprovalsForSealConstruction)
```

Comment:
separated logic for creating inbound queues for trusted inputs from message handler for untrusted inputs